bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/__init__.py

```diff
@@ -0,0 +1,161 @@
+python_classes = [Factory, Dataloader, CrossValidation]
+
+# create a dictionary for convenient access to python classes
+python_classes_dict = dict()
+for python_class in python_classes:
+    python_classes_dict[python_class.__name__] = python_class
+
+
+def get_defined_attributes(klass, consider_base_classes=False):
+    """
+    Get all class attributes that are defined in a given class or optionally in
+    any of its base classes (except for `object`).
+    """
+
+    blacklist = ["__dict__", "__doc__", "__hash__", "__module__", "__weakref__"]
+
+    if not consider_base_classes:
+        return sorted([attr for attr in klass.__dict__.keys() if attr not in blacklist])
+
+    # get a list of this class and all its base classes, excluding `object`
+    method_resolution_order = klass.mro()
+    if object in method_resolution_order:
+        method_resolution_order.remove(object)
+
+    def is_defined(funcname):
+
+        if funcname in blacklist:
+            return False
+
+        in_any_dict = False
+
+        for mro_class in method_resolution_order:
+            if funcname in mro_class.__dict__:
+                in_any_dict = True
+
+        return in_any_dict
+
+    return sorted([attr for attr in dir(klass) if is_defined(attr)])
+
+
+def is_staticmethod_py2(klass, func_name):
+    """Check if the function with name `func_name` of a class is a static method in Python 2."""
+
+    return type(getattr(klass, func_name)).__name__ == "function"
+
+
+def is_classmethod(klass, func):
+    if hasattr(func, "__self__"):
+        return func.__self__ == klass
+    return False
+
+
+def rebind_attribute(to_class, from_class, func_name):
+    """
+    Bind the instance method `from_class.func_name` also to class `to_class`.
+    """
+
+    import sys
+
+    from_method = getattr(from_class, func_name)
+
+    if is_classmethod(from_class, from_method):
+        # the @classmethod case
+        to_method = classmethod(from_method.__func__)
+    elif sys.version_info >= (3, 0):
+        # any other case in Python 3 is trivial
+        to_method = from_method
+    elif isinstance(from_method, property):
+        # the @property case in Python 2
+        to_method = from_method
+    elif is_staticmethod_py2(from_class, func_name):
+        # the @staticmethod case in Python 2
+        to_method = staticmethod(from_method)
+    else:
+        # the instance method case in Python 2
+        import new
+
+        to_method = new.instancemethod(from_method.__func__, None, to_class)
+
+    setattr(to_class, func_name, to_method)
+
+
+def make_func_name_orig(func_name):
+    """Return the name that we will give to the original cppyy function."""
+    # special treatment of magic functions, e.g.: __getitem__ > _getitem
+    if func_name.startswith("__") and func_name.endswith("__"):
+        func_name = func_name[2:-2]
+
+    return "_" + func_name
+
+
+@pythonization(class_name="TMVA", is_prefix=False)
```

Comment:
I believe this attempts to pythonize the whole TMVA namespace? That can't work, since the `pythonization` decorator applies to classes. So here you should something like `@pythonization("SomeClass", ns="TMVA")`